src/controllers/MagicLoginController.php

```diff
@@ -87,20 +88,64 @@ public function actionLogin()
     {
         $this->requirePostRequest();
 
-        $emailOrUsername = Craft::$app
+        $email = Craft::$app
             ->getRequest()
-            ->getRequiredParam('emailOrUsername');
-    
-        $result = MagicLogin::$plugin
-            ->magicLoginAuthService
-            ->createMagicLogin($emailOrUsername);
+            ->getRequiredParam('email');
 
-        if (!$result) {
-            // TODO: Trigger some kind of error or say if email exists then email has 
-            // been sent (this is based on configuration).
+        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            // TODO: Set this to be configurable.
+            $this->setFailFlash(\Craft::t('magic-login', 'Please enter a valid email address.'));
+            return;
+        }
+        
+        $link = MagicLogin::$plugin
+            ->magicLoginAuthService
+            ->createMagicLogin($email);
+
+        if (!$link) {
+            // Although this looks to be successful in reality we don't
```

Comment:
If we couldn't generate a link (likely because a user doesn't exist). This request is actually unsuccessful however to a user on the frontend we are making it look successful to prevent someone validating if email address' exist on the website.